### PR TITLE
first whack documenting `[` behavio(u)r for `tbl_df()` objects

### PR DESCRIPTION
**Pull request by ijlyttle**
In response to Issue #1345

If thought to be useful, I can also try to put something in the man page for tbl_df and/or data_frame. 

Up to you (of course).

### CODE REVIEW

File: vignettes/data_frames.Rmd [hunk 1]
```diff
@@ -53,6 +53,22 @@ library(dplyr)
     ```{r}
     data_frame(x = 1:5) %>% class()
     ```
+
+  * It changes the subsetting behaviour to make the use of `[` consistent.
+    For a `tbl_df()` object, subsetting by `[` will always return a `tbl_df()`
+    object; whereas subsetting by `[[` will always return its (subsetted)
```
Comment (hadley) [inline vignettes/data_frames.Rmd]: I'd just say a column

File: vignettes/data_frames.Rmd [hunk 2]
```diff
@@ -53,6 +53,22 @@ library(dplyr)
     ```{r}
     data_frame(x = 1:5) %>% class()
     ```
+
+  * It changes the subsetting behaviour to make the use of `[` consistent.
```
Comment (hadley) [inline vignettes/data_frames.Rmd]: Maybe "It changes the behaviour of `[` to always return the same type of object."

File: vignettes/data_frames.Rmd [hunk 3]
```diff
@@ -53,6 +53,22 @@ library(dplyr)
...
+    For a `tbl_df()` object, subsetting by `[` will always return a `tbl_df()`
+    object; whereas subsetting by `[[` will always return its (subsetted) 
+    contents.
+    
+    The user should be aware of one subsetting case where a `tbl_df()` object
```
Comment (hadley) [inline vignettes/data_frames.Rmd]: The user -> you

**Comment by ijlyttle** [comment]
I don't know if you want me to keep the second sentence, beginning "For a `tbl_df()` object,"

I'm sure you'll let me know.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**